common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -427,7 +427,10 @@ static ByteBuffer reallocateDirectNoCleaner(ByteBuffer buffer, int capacity) {
     }
 
     static ByteBuffer allocateDirectNoCleaner(int capacity) {
-        return newDirectBuffer(UNSAFE.allocateMemory(capacity), capacity);
+        // Calling malloc with capacity of 0 may return a null ptr or a memory address that can be used.
+        // Just use 1 to make it safe to use in all cases:
+        // See: http://pubs.opengroup.org/onlinepubs/009695399/functions/malloc.html
+        return newDirectBuffer(UNSAFE.allocateMemory(Math.max(1, capacity)), capacity);
```

Comment:
won't this result in a buffer that has a longer length than it's capacity?

Reply:
@carl-mastrangelo generally speaking yes... that said `capacity()` will return the original `capacity()` so its not a problem.

Same is true in OpenJDK as well:
http://hg.openjdk.java.net/jdk-updates/jdk10u/file/523e40ed3fc7/src/java.base/share/classes/java/nio/Direct-X-Buffer.java.template#l122